code/go/0chain.net/blobbercore/reference/ref.go

```diff
@@ -324,55 +426,55 @@ func (fr *Ref) GetFileHashData() string {
 }
 
 func (fr *Ref) CalculateFileHash(ctx context.Context, saveToDB bool) (string, error) {
+
 	fr.Hash = encryption.Hash(fr.GetFileHashData())
-	fmt.Println("file hash", fr.Path, fr.Hash)
 	fr.NumBlocks = int64(math.Ceil(float64(fr.Size*1.0) / float64(fr.ChunkSize)))
 	fr.PathHash = GetReferenceLookup(fr.AllocationID, fr.Path)
 	fr.PathLevel = len(GetSubDirsFromPath(fr.Path)) + 1
 	fr.LookupHash = GetReferenceLookup(fr.AllocationID, fr.Path)
+
 	var err error
 	if saveToDB {
-		err = fr.Save(ctx)
+		//err = fr.Save(ctx)
+		err = fr.SaveFile(ctx)
 	}
 	return fr.Hash, err
 }
 
 func (r *Ref) CalculateDirHash(ctx context.Context, saveToDB bool) (string, error) {
 	// empty directory, return hash directly
+
 	if len(r.Children) == 0 && !r.childrenLoaded {
 		return r.Hash, nil
 	}
 	sort.SliceStable(r.Children, func(i, j int) bool {
```

Comment:
@abouroumine I think if we do `db.Order("path", "level")` then this sorting is redundant. Do you think so? 
Also the comparison it makes with `LookupHash` is questionable. We cannot order path with lookuphashes

Reply:
I think you are right the sorting is indeed redundant. and can please explain more the difference between db.Order("path","level") than db.Order("level","path").